app/vue/contexts/CompetitionsPageContext.js

```diff
@@ -74,43 +75,112 @@ export default class CompetitionsPageContext extends BaseFuroContext {
    * @this {T}
    */
   setupComponent () {
-    const route = useRoute()
-    const currentPageComputed = computed(() => (
-      isNaN(Number(route.query.page))
-        ? 1
-        : Number(route.query.page)
-    ))
-
     this.graphqlClientHash
       .competitionStatistics
       .invokeRequestOnMounted({
         hooks: this.competitionStatisticsLauncherHooks,
       })
 
-    this.watch(
-      () => route.query.page,
-      () => this.graphqlClientHash
-        .competitions
-        .invokeRequestOnEvent({
-          variables: {
-            ...this.defaultCompetitionsVariables,
-            input: {
-              pagination: {
-                ...this.defaultCompetitionsVariables.input.pagination,
-                offset: (currentPageComputed.value - 1) * PAGINATION.LIMIT,
-              },
+    const queryTitle = this.extractQueryTitle()
+    const currentPage = this.extractCurrentPage()
+
+    this.graphqlClientHash
+      .competitions
+      .invokeRequestOnMounted({
+        variables: {
+          ...this.defaultCompetitionsVariables,
+          input: {
+            title: queryTitle,
+            pagination: {
+              ...this.defaultCompetitionsVariables.input.pagination,
+              offset: (currentPage - 1) * PAGINATION.LIMIT,
             },
           },
-          hooks: this.graphqlRequestHooks,
-        }),
-      {
-        immediate: true,
+        },
+        hooks: this.graphqlRequestHooks,
+      })
+
+    this.watch(
+      () => this.extractCurrentPage(),
+      async () => {
+        await this.fetchCompetitions()
       }
     )
 
     return this
   }
 
+  /**
+   * Fetch competitions
+   *
+   * @param {{
+   *   title?: string
+   *   statusId?: number
+   * }} [params] - Parameters.
+   * @returns {Promise<void>}
+   */
+  async fetchCompetitions ({
+    title,
+    statusId,
+  } = {}) {
+    await this.router.replace({
+      query: {
+        ...this.route.query,
+        title: title || REMOVED_QUERY_VALUE,
+      },
+    })
```

Comment:
@hecticme 

* Original code
```
    await this.router.replace({
      query: {
        ...this.route.query,
        title: title || REMOVED_QUERY_VALUE,
      },
    })
```

Two points

# (1) Do not use short circuit evalution anyway

# (2) Separate two logics in two

Original code includes two logics in one statement
It violates ORT coding charter "(7) Keep Everything One by One"

```suggestion
    const query = title
      ? {
        ...this.route.query,
        title,
      }
      : this.route.query

    await this.router.replace({
      query,
    })
```

Reply:
@hecticme 

I fix the kicking out title logic

```suggestion
    const {
      title,
      ...restQuery
    } = this.route.query
    
    const query = title
      ? {
        ...restQuery,
        title,
      }
      : restQuery

    await this.router.replace({
      query,
    })
```

Reply:
@hecticme 

More better

```suggestion
    const query = this.buildQuery({ // <--- 👀
      title,
    })
    
    await this.router.replace({
      query,
    })
```

Reply:
@StewEucen 

I considered moving the rest query logic into a dedicated instance method, as suggested, but ran into an issue with scaling the code.

We probably want the buildQuery method to work with an object with multiple properties.
```js
doSomething () {
  const query = this.buildQuery({
    title,
    somethingElse,
  })
}

/**
 * ⚠️ NOT ACTUAL PARAMETER TYPE. JUST FOR DEMONSTRATION PURPOSE.
 * @param {Record<string, LocationQueryValue>} query.
 */
buildQuery (query) {
  const listedQueries = Object.keys(query)

  const restQuery = Object.fromEntries(
    Object.entries(this.route.query)
      .filter(([key]) => !listedQueries.includes(key))
  )

  const filteredQuery = Object.fromEntries(
    Object.entries(this.route.query)
      .filter(([_, value]) => value)
  )

  return {
    ...restQuery,
    ...filteredQuery,
  }
}
```

It looks a bit complicated so I wonder if it's ok.

Of course, I can write a dedicated method just for `title`. More simple but with trade-off in scalability 🤔 

```
buildTitleReplacementQuery ({
  title,
}) {
    const {
      title: _,
      ...restQuery
    } = this.route.query
    const replacementQuery = title
      ? {
        ...restQuery,
        title,
      }
      : restQuery
}
```

Reply:
@hecticme 

We could apply YAGNI principle to the issue

https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

1. `#fetchCompetitions()` does not generalized code now, even if requires it in future
2. Since the responsibility to generate value of `query` is not for `#fetchCompetitions()`, it is good to cut out the logic to replace `titile` only to `#buildTitleReplacementQuery()`